python/cppyy/numba_ext.py

```diff
@@ -379,7 +495,11 @@ def typeof_scope(val, c, q = Qualified.default):
     class ImplClassType(CppClassNumbaType):
         pass
 
-    cnt = ImplClassType(val, q)
+    if is_instance:
```

Comment:
Why is `q` above set to `Qualified.default` and `is_instance` set to a separate variable? Would be nice to have all the necessary information in`q` only and pass that around.

Reply:
I initially went ahead with that approach but code segments like
`` try:
        return scope_numbatypes[q][val]`` and a significant part of control flow in both ```typeof_scope``` and ```cppclass_getattr_impl``` required refactoring for both ```Qualified.default``` and ```Qualified.instance```
This way I can obtain the passed cppinstance_val and the rest of the flow remains the same by resetting q from instance to default in typeof_scope. will definitely look into better approaches